enhancements/user-defined-default-ingress-controller.md

```diff
@@ -0,0 +1,273 @@
+      provisioned.
+    * DNS records for the old cloud load balancer are destroyed and then
+      re-created for the new load balancer.
+
+If this proposal is implemented, during installation the user can provide the
+following IngressController manifest which satisfies the constraint as well as
+eliminating disruption and unnecessary (and possibly expensive) cloud resource
+provisioning.
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  namespace: openshift-ingress-operator
+  name: default
+spec:
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+```
+
+#### Pre-configuring shards
+
+>As an OpenShift administrator, I want to the default IngressController to only
+>admits OpenShift's default Routes
+
+To accomplish this goal today, users must:
+
+1. Install a cluster which by default produces an IngressController which will
+   admit any Route resource in any namespace.
+2. Update the default IngressController to specify Route selector criteria which
+   exclude user defined Routes (where user defined Route is a Route labeled to
+   indicate the Route is user-defined).
+
+The update procedure has at least two notable drawbacks:
+
+1. Changing the IngressController will result in a rollout (which today can be
+   disruptive).
+2. The IngressController may already have admitted some Routes which would have
+   been excluded had the IngressController been created with the desired
+   selection criteria.
+
+If this proposal is implemented, during installation the user can provide an
+IngressController manifest which has the desired Route selection criteria,
+eliminating both drawbacks.
+
+### Implementation Details
+
+The IngressController resource is a cluster-scoped Custom Resource Definition.
+For the cluster to support creation of the default IngressController resource,
+the following must be true:
+
+1. The IngressController Custom Resource Definition must be persisted and
+   admitted into the cluster.
+2. The `openshift-ingress-operator` namespace must exist to contain the
+   `default` IngressController resource.
+
+The Custom Resource Definition and Namespace manifests are part of the Ingress
+Operator image, and are normally created and managed by the [Cluster Version Operator](https://github.com/openshift/cluster-version-operator). Creation order of these resources
+is significant, and it is unsafe for users to reorder or modify them.
+
+To ensure a safe and supportable path for creating these resources at
+installation time and outside the context of the CVO, the proposed Ingress
+Operator `render` command emits these manifests with the correct order and
+content, and users are warned that modifying them results in an unsupportable
+cluster.
+
+The user provides _only_ the IngressController resource.
+
+Constraints:
+
+* **After installation has completed, OpenShift has no memory of the
```

Comment:
This is the only questionable part of this design.

Reply:
> This is the only questionable part of this design.

thinking on this further, I think that if a user deletes the default ingress operator, their intent is to remove the default ingress operator. We don't need to recreate.  Rendering it all in the installer avoids the need for an operator to auto recreate

Reply:
in the future, we could have an option that disables auto-creation in ingress operator.